src/Jackalope/NodeType/NodeProcessor.php

```diff
@@ -71,18 +72,18 @@ class NodeProcessor
     private $autoLastModified;
 
     /**
-     * @var array List of namespaces known to the current session. Keys are prefix, values are URI.
+     * @var ArrayObject List of namespaces known to the current session. Keys are prefix, values are URI.
      */
-    private $namespaces = array();
+    private $namespaces;
 
     /**
-     * @param string  $userId           ID of the connected user
-     * @param array   $namespaces       List of namespaces in the current session. Keys are prefix, values are URI.
-     * @param boolean $autoLastModified Whether the last modified property should be updated automatically
+     * @param string      $userId           ID of the connected user
+     * @param ArrayObject $namespaces       List of namespaces in the current session. Keys are prefix, values are URI.
+     * @param bool        $autoLastModified Whether the last modified property should be updated automatically
      */
     public function __construct(
         $userId,
-        $namespaces = array(),
+        ArrayObject $namespaces,
```

Comment:
this makes $namespaces not optional anymore. but you verified usage of NodeProcessor and the namespaces are always passed in?